src/repo.rs

```diff
@@ -811,6 +825,30 @@ impl Repository {
         }
     }
 
+    /// Create object database reading stream
+    pub fn odb_reader(&self, oid: Oid) -> Result<OdbReader, Error> {
+        let mut out = ptr::null_mut();
+        unsafe {
+            let mut db = ptr::null_mut();
+            try_call!(raw::git_repository_odb(&mut db, self.raw()));
+            try_call!(raw::git_odb_open_rstream(&mut out, db, oid.raw()));
+            raw::git_odb_free(db);
```

Comment:
Of course, initially I planned to introduce ODB objects, but after looking the libgit2 sources I seen that the common practice is accessing to ODB by weakptr because the ODB associated with repository. For reference you can look the source of  [git_blob_create_frombuffer](https://github.com/libgit2/libgit2/blob/master/src/blob.c#L62) function. I used `git_repository_odb__weakptr` like here. It works for me, but have the compilation error on your CI (may be it related to different versions of libgit2).